src/System.Buffers.Experimental/System/Buffers/Multispan.cs

```diff
@@ -0,0 +1,213 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Buffers
+{
+    /// <summary>
+    /// A collection of pooled arrays.
```

Comment:
This is confusing...what scenario does this allow that the regular pool does not? It looks like (and this comment specifies) that this is just a collection of different size, pooled ```T```s.